### ISSUE TITLE
`tjson`: Support `null`

### ISSUE THREAD

**Issue by rumyantseva** [FerretDB/FerretDB#906]
This task is a part of #683 epic.
This task should be taken after #899 is done.

🎯 The goal of the task is to support `null` type for `tjson`.
💡 One possible option to solve this task is to write integration tests first.

### Null in Tigris

This is how `null` values [work in Tigris](https://docs.tigrisdata.com/overview/schema#null-values):
- Primary key's fields can't be set as `null`.
- Any other field can be explicitly set as `null` in the JSON payload.
- If the field is set as `null` it will be stored as `null`.
- If the field is not set in the JSON payload, it is not stored in Tigris.
- If the field doesn't have existing schema but the value is set as null, the field will not occur in the schema

### Readiness checklist

- [x] Add `null`-related unit tests to the `tjson` package. Any field could be nullable if it's not a part of the primary key.
- [x] Add a nullable field to one of the test cases in the `document_test.go` file.
- [ ] The implemented cases (when `null` is possible and when `null` is not possible) need to be covered with integration tests. Dance tests are needed for the cases that are Tigris-specific.
    - [ ] Cases to cover:
        - [ ] Insert, upsert, update, unset
        - [ ] Queries
- [ ] Add special tests for the case when the field is not set in the payload (insert, upsert, queries etc)
    - [ ] Three documents in the collection: one has the field `A` and it's set as `string`, second has the field `A` and it's set as `null`, third doesn't have the field `A`.
- [x] Check that all shared data sets have Tigris schemas set (including composites!)

**Comment by rumyantseva** [comment]
ℹ️ Update operation is not in scope as currently null values are not supported for updates in Tigris, see a separate issue https://github.com/FerretDB/FerretDB/issues/1019.

**Comment by rumyantseva** [comment]
- Check update case if the field is not presented in the payload

**Comment by rumyantseva** [comment]
Remove from the Sprint for now as there are too many corner cases
